girder_worker/plugins/docker/executor.py

```diff
@@ -118,6 +118,26 @@ def _docker_gc(tempdir):
     return subprocess.Popen(args=(script,), env=env)
 
 
+def validate_task_outputs(task_outputs):
+    """
+    This is called prior to fetching inputs to make sure the output specs are
+    valid. Outputs in docker mode can result in side effects, so it's best to
+    make sure the specs are valid prior to fetching.
+    """
+    for name, spec in task_outputs.iteritems():
+        if spec.get('target') == 'filepath':
+            if 'path' not in spec:
```

Comment:
My bad. Forgot to git pull.

Now that i am modifying my code to provide the path, i feel the binding spec is more appropriate.
I feel the taskSpec is supposed to say `this task will provide an output file of type blah`. And the bindingSpec is supposed to say `this output file should be stored at xyz folder in girder with name abc`